pipeline/authn/authenticator_cookie_session_test.go

```diff
@@ -58,7 +58,7 @@ func TestAuthenticatorCookieSession(t *testing.T) {
 		t.Run("description=should pass through method, path, and headers to auth server", func(t *testing.T) {
 			testServer, requestRecorder := makeServer(200, `{"subject": "123"}`)
 			err := pipelineAuthenticator.Authenticate(
-				makeRequest("PUT", "/users/123?query=string", map[string]string{"sessionid": "zyx"}, ""),
+				makeRequest("PUT", "/users/123", "query=string", map[string]string{"sessionid": "zyx"}, ""),
 				session,
 				json.RawMessage(fmt.Sprintf(`{"check_session_url": "%s"}`, testServer.URL)),
```

Comment:
Could you please add a test which uses a `check_session_url` that has a query parameter and observes the behaviour dependent on preserve_query?

Reply:
True! It should be covered now :)